### PR TITLE
[14.0][ADD] shopinvader_mass_mailing 📨

### PR DESCRIPTION
**Pull request by ivantodorovich**
Exposes a mailing lists service to allow visitors to subscribe / unsubscribe from them.

### CODE REVIEW

File: shopinvader_mass_mailing/services/mailing_list.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+from odoo.addons.base_rest import restapi
+from odoo.addons.component.core import Component
+
+
+class MailingListsService(Component):
+    """Shopinvader Service to manage mailing list subscriptions."""
+
+    _name = "shopinvader.mailing.lists.service"
+    _inherit = "base.shopinvader.service"
+    _usage = "mailing.lists"
```
Comment (simahawk) [inline shopinvader_mass_mailing/services/mailing_list.py]:
```suggestion
    _usage = "mailing_lists"
```

File: shopinvader_mass_mailing/services/mailing_list.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+    def _get_base_search_domain(self):
+        return [("is_public", "=", True)]
+
+    def _json_parser(self):
+        res = [
+            "id",
+            "name",
+        ]
+        # If the user is authenticated, add the subscription status
+        if self.work.partner_user:
```
Comment (simahawk) [inline shopinvader_mass_mailing/services/mailing_list.py]:
```suggestion
        if self._is_logged_in():
```

File: shopinvader_mass_mailing/tests/common.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.addons.shopinvader.tests.common import CommonCase
+
+
+class CommonMassMailingCase(CommonCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
```
Comment (simahawk) [inline shopinvader_mass_mailing/tests/common.py]: tracking is disabled on CommonCase setup

**Review by simahawk — COMMENTED** [review]
LG overall

**Comment by ivantodorovich** [comment]
Done!

Tests are failing due to `openupgradelib` though:
```
  Running command git clone -q https://github.com/OCA/openupgradelib.git /tmp/pip-install-gcz7x0j0/openupgradelib_9d8187731d0046faba61f12ced52508d
  Cloning https://github.com/OCA/openupgradelib.git to /tmp/pip-install-gcz7x0j0/openupgradelib_9d8187731d0046faba61f12ced52508d
  Resolved https://github.com/OCA/openupgradelib.git to commit 871058afeb6dbd9b34314108b4a8ca6f3e0a9d9f
    ERROR: Command errored out with exit status 1:
     command: /opt/odoo-venv/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-gcz7x0j0/openupgradelib_9d8187731d0046faba61f12ced52508d/setup.py'"'"'; __file__='"'"'/tmp/pip-install-gcz7x0j0/openupgradelib_9d8187731d0046faba61f12ced52508d/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-zshxn1wq
         cwd: /tmp/pip-install-gcz7x0j0/openupgradelib_9d8187731d0046faba61f12ced52508d/
    Complete output (7 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-gcz7x0j0/openupgradelib_9d8187731d0046faba61f12ced52508d/setup.py", line 5, in <module>
        import openupgradelib
      File "/tmp/pip-install-gcz7x0j0/openupgradelib_9d8187731d0046faba61f12ced52508d/openupgradelib/__init__.py", line 14, in <module>
        from importlib_metadata import version, PackageNotFoundError
    ModuleNotFoundError: No module named 'importlib_metadata'
```

Seems related to https://github.com/OCA/openupgradelib/pull/243

**Comment by simahawk** [comment]
yeah, I got the same issue and reported here https://github.com/shopinvader/odoo-shopinvader/issues/1111 
CC @sbidoul

**Comment by sbidoul** [comment]
Please restart test jobs

**Comment by sbidoul** [comment]
BTW, this is also a set of REST services that would benefit of being independent of shopinvader (like helpdesk, event, etc).
I think the only abstraction we need is a mechanism to determine the authenticated `res.partner`.
Then these can go into OCA to benefit and attract broader community.

**Comment by simahawk** [comment]
I agree, but we could add this to the roadmap for v15.

**Comment by sbidoul** [comment]
Ah, yes, v15 :rofl:

**Comment by sebastienbeau** [comment]
@PierrickBrun is working on helpdesk case. When he will have a working version he can help on this

**Comment by simahawk** [comment]
TODO: review sudo changes

**Comment by ivantodorovich** [comment]
rebased to solve conflicts

**Review by sebastienbeau — APPROVED** [review]
LGTM (code review)

**Pull request closed without merging**